### PR TITLE
[WC-2535]: Fix barcodescanner error on module load

### PR DESCRIPTION
**Pull request by HedwigAR**
### Pull request type

Bug fix (non-breaking change which fixes an issue)

---

### Description

Currently an issue exists where the module won't load because the reference to the canvas it needs is undefined. This PR changes this reference from a state to a ref, which ensures the reference can be found in the async process.

### What should be covered while testing?
If the camera view appears with crop enabled, it means the canvas reference is not undefined, so the issue is fixed. While this shouldn't impact other features in the widget, it would be good to double-check they all work so we won't have to unpublish the widget again.

### CODE REVIEW

**Review by gjulivan — DISMISSED** [review]

File: packages/pluggableWidgets/barcode-scanner-web/src/components/BarcodeScanner.tsx [hunk 1]
```diff
@@ -112,7 +112,7 @@ export function BarcodeScanner({
             showMask={showMask}
             canvasMiddleMiddleRef={ref => {
                 if (ref !== null) {
-                    setCanvasMiddleRef(ref);
+                    canvasMiddleRef.current = ref;
                 }
             }}
```
Comment (iobuhov) [inline packages/pluggableWidgets/barcode-scanner-web/src/components/BarcodeScanner.tsx:117]: I would suggest instead, change prop type on line 15, to React.RefObject and pass it as it is without using callback. Callback as a ref it's legacy react approach and in this case it not add any value. Just use ref as it is.

**Review by iobuhov — CHANGES_REQUESTED** [review]

File: packages/pluggableWidgets/barcode-scanner-web/src/components/BarcodeScanner.tsx (hunk 1 above)
Comment (HedwigAR) [reply packages/pluggableWidgets/barcode-scanner-web/src/components/BarcodeScanner.tsx:117]: Alright, thanks for the suggestion, I thought doing that made it too cluttered, and just having the object there would be nicer. I'll change it.

Comment (HedwigAR) [reply packages/pluggableWidgets/barcode-scanner-web/src/components/BarcodeScanner.tsx]: I changed it, let me know if this is right

**Review by iobuhov — DISMISSED** [review]
LGTM

**Review by iobuhov — APPROVED** [review]
LGTM

**Review by leonardomendix — APPROVED** [review]

**Pull request merged**